core/cap-TBD.md

```diff
@@ -0,0 +1,175 @@
+## Preamble
+
+```
+CAP: TBD
+Title: Separate host functions to extend the TTL for contract instance and contract code
+Working Group:
+    Owner: Anup Pani <@anupsdf>
+    Authors: Tommaso De Ponti <@heytdep>
+    Consulted: Leigh McCulloch <@leighmcculloch>, Dmytro Kozhevin <@dmkozh>
+Status: Draft
+Created: 2023-03-06
+Discussion: TBD
+Protocol version: TBD
+```
+
+## Simple Summary
+
+Allow extending the Time To Live (TTL) for contract instance and contract code with separate Soroban smart contract host functions.
+
+## Working Group
+
+This change was authored by Tommaso De Ponti, with input from the consulted individuals mentioned at the top of this document.
+
+#### Semantic protocol changes
+
+Adding two Soroban smart contract host functions.
+
+## Motivation
+
+Currently, Soroban smart contract system has a host function, extend_contract_instance_and_code_ttl_from_contract_id, that extends the [TTL](cap-0046-12.md) of both contract instance and contract code ledger entries by the same value. In decentralized contracts, the contract can extend its own lifetime from within the code with certain thresholds with the idea that the cost is distributed among its users. 
+
+Extending the TTL for contract code entries are very expensive due to the large binary sizes that occupy the ledger. There are numerous situations where a contract code entry is referenced by multiple contract instances. Thus allowing to extend them separately would enable implementing a more efficient lifetime extension logic. 
+
+For example, a liquidity pool contract can be used by thousands of actively bumped contract instances. The contract instance of a single pool contract can be bumped by the users of that contract, but the contract code entry can be bumped by the users of all the pool contracts. So, when extending the lifetime of a contract instance, extending the lifetime of contract code separately and slightly less would make up for a better distribution of the fees across the network.
+
+### Goals Alignment
+
+This CAP is aligned with the following Stellar Network Goals:
+
+  - The Stellar Network should make it easy for developers of Stellar projects to create highly
+  usable products.
+
+## Abstract
+
+This CAP introduces these two new Soroban smart contract host functions to bump the TTL of contract code and instance:
+1. extend_contract_instance_ttl to extend the contract instance's TTL
+2. extend_contract_code_ttl to extend the contract code's TTL
+
+## Specification
+
+Two new functions `extend_contract_instance_ttl` and `extend_contract_code_ttl` with export name `c` and `d` are added to 
+the Soroban environment's exported interface.
```

Comment:
Which module are they being added to? The exports should exist within a module.

Reply:
Good catch! I added module `l` for ledger in the latest commit. That's the module that contains the existing TTL functions.

Reply:
Also changed the export names to 10 & 11 since existing TTL functions end at 9.